res/controllers/Reloop-Beatmix-2-4-scripts.js

```diff
@@ -0,0 +1,718 @@
+        engine.scratchEnable(deck, 1600, 33 + 1 / 3, alpha, beta);
+    } else {
+        engine.scratchDisable(deck);
+    }
+};
+
+ReloopBeatmix24.WheelTurn = function(channel, control, value, status, group) {
+    var newValue = value - 64;
+    var deck = parseInt(group.substr(8, 1), 10);
+
+    // In either case, register the movement
+    if (engine.isScratching(deck)) {
+        engine.scratchTick(deck, newValue); // Scratch!
+    } else {
+        engine.setValue(group, 'jog', newValue); // Pitch bend
+    }
+};
+
+// Led Feedback functions
+// ========================================================
+ReloopBeatmix24.AllJogLEDsToggle = function(deck, state, step) {
+    var step = typeof step !== 'undefined' ? step : 1; // default value
+    for (var j = 0x30; j <= 0x3F; j += step) {
+        midi.sendShortMsg(deck, j, state);
+    }
+};
+
+ReloopBeatmix24.deckloaded = function(value, group, control) {
+    var i;
+    switch (group.substr(1, 7)) {
+        case "Channel":
+            var channelRegEx = /\[Channel(\d+)\]/;
+            var channelChan = parseInt(channelRegEx.exec(group)[1]);
+            if (channelChan <= 4) {
+                midi.sendShortMsg(0x90 + channelChan, 0x50, value ? ON :
+                    OFF);
+            }
+            break;
+        case "Sampler":
+            var samplerRegEx = /\[Sampler(\d+)\]/;
+            var samplerChan = parseInt(samplerRegEx.exec(group)[1]);
+            if (samplerChan <= 8) { // We only handle 8 samplers (1 per pad)
+                for (i = 0x91; i <= 0x94; i++) {
+                    // PAD1 Mode A
+                    midi.sendShortMsg(i, 0x08 - 1 + samplerChan, value ?
+                        RED : OFF);
+                    // SHIFT+PAD1 Mode A
+                    midi.sendShortMsg(i, 0x48 - 1 + samplerChan, value ?
+                        RED : OFF);
+                    if (samplerChan <= 4) { // Handle first 4 samplers in split mode
+                        // PAD5 Mode A+B (sampler 1 in split mode)
+                        midi.sendShortMsg(i, 0x14 - 1 + samplerChan, value ?
+                            RED : OFF);
+                        // SHIFT+PAD5 Mode A+B (sampler 1 in split mode)
+                        midi.sendShortMsg(i, 0x1C - 1 + samplerChan, value ?
+                            RED : OFF);
+                    }
+                }
+            }
+            break;
+    }
+};
+
+ReloopBeatmix24.SamplerPlay = function(value, group, control) {
+    var samplerRegEx = /\[Sampler(\d+)\]/;
+    var samplerChan = parseInt(samplerRegEx.exec(group)[1]);
+    if (samplerChan <= 8) { // We only handle 8 samplers (1 per pad)
+        var ledColor;
+        if (value) {
+            ledColor = VIOLET;
+        } else {
+            ledColor = engine.getValue(group, "track_samples") ? RED : OFF;
+        }
+
+        // We need to switch off pad lights before changing color otherwise
+        // VIOLET to RED transition does not work well. (???)
```

Comment:
Strange. What do you mean by "does not work well"? Does turning it off in between make the light flicker?